common/src/main/java/com/wynntils/functions/CharacterFunctions.java

```diff
@@ -112,6 +113,88 @@ public String getValue(String argument) {
         }
     }
 
+    public static class LiquidEmeraldFunction extends DependantFunction<Integer> {
+        @Override
+        public Integer getValue(String argument) {
+            int ems = PlayerInventoryModel.getCurrentEmeraldCount();
+            return (ems - (ems % 4096)) / 4096;
+        }
+
+        @Override
+        public List<String> getAliases() {
+            return List.of("le");
+        }
+
+        @Override
+        public List<Class<? extends Model>> getModelDependencies() {
+            return List.of(PlayerInventoryModel.class);
+        }
+    }
+
+    public static class EmeraldBlockFunction extends DependantFunction<Integer> {
+        @Override
+        public Integer getValue(String argument) {
+            int ems = PlayerInventoryModel.getCurrentEmeraldCount();
+            return (ems % 4096 - (ems % 4096 % 64)) / 64;
+        }
+
+        @Override
+        public List<String> getAliases() {
+            return List.of("eb");
+        }
+
+        @Override
+        public List<Class<? extends Model>> getModelDependencies() {
+            return List.of(PlayerInventoryModel.class);
+        }
+    }
+
+    public static class EmeraldsFunction extends DependantFunction<Integer> {
+        @Override
+        public Integer getValue(String argument) {
+            return PlayerInventoryModel.getCurrentEmeraldCount() % 4096 % 64;
```

Comment:
No need to do both %, the latter is enough

Reply:
ah because 4096 is multiple of 64